src/transformer/descriptor/mock/mockProperties.ts

```diff
@@ -19,21 +19,21 @@ export function GetMockPropertiesFromSymbol(propertiesSymbol: ts.Symbol[], signa
 
 export function GetMockPropertiesFromDeclarations(list: ReadonlyArray<PropertyLike>, signatures: ReadonlyArray<SignatureLike>, scope: Scope): ts.CallExpression {
   const propertiesFilter: PropertyLike[] = list.filter((member: PropertyLike) => {
-    const hasModifiers: boolean = !!member.modifiers;
+    const { modifiers = ts.createNodeArray([]) }: PropertyLike = member;
```

Comment:
these changes are not necessary. Could we revert these changes in this PR?

Reply:
Sure. I guess doing an unnecessary `ts.createNodeArray([])` call was a bit of a stretch. The goal is to type guard modifiers or tell the type checker that we know what we're doing and add an exclamation mark on `modifiers` in the return statement.

Reply:
On that note, the ESLint config disallows the use of `!` so we would have to cast it instead. I think type guarding is cleaner though and more explicit. `isTruthyValue` here can be used in more places than this, however, the requirement of variable typing would be a bit tedious (since `hasModifiers` is now a function).

```
const isTruthyValue = <T>(arg: T): arg is NonNullable<T> => {
  return !!c;
};

const hasModifiers = isTruthyValue;

if (!hasModifiers(member.modifiers)) {
  return true;
}

return member.modifiers.filter((modifier: ts.Modifier) => modifier.kind === ts.SyntaxKind.PrivateKeyword).length === 0;
```

Reply:
I see, let's simply assign modifiers to a variable. 

```ts
const modifiers: ts.NodeArray<ts.Modifier> | undefined = member.modifiers;

    if (IsTypescriptType(member)) { // This is a current workaround to safe fail extends of TypescriptLibs
      return false;
    }

    if (!modifiers) {
      return true;
    }

    return modifiers.filter((modifier: ts.Modifier) => modifier.kind === ts.SyntaxKind.PrivateKeyword).length === 0;
```